packages/lexical-list/src/LexicalListNode.ts

```diff
@@ -49,6 +50,26 @@ export type SerializedListNode = Spread<
 
 export type ListType = 'number' | 'bullet' | 'check';
 
+export type ListStyleType =
+  | undefined // default
+  | (string & object) // string means custom style
```

Comment:
```suggestion
  // eslint-disable-next-line @typescript-eslint/ban-types
  | (string & {}) // string means custom style
```

object and {} don't mean the same thing, using object makes it not accept arbitrary strings because `string & object = never`

[Playground Link](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAGlC8UDOwBOBLAdgcygMigHsAjAKwgGNgBuAKAHp6pmA9AfltEigE0Fk0WXAQDeAXzqNmUdrVpA)